src/index.js

```diff
@@ -63,6 +64,10 @@ function IpfsAPI (host_or_multiaddr, port) {
       opts = {}
     }
 
+    if (typeof files === 'string' && files.startsWith('http')) {
+      files = request(files)
```

Comment:
The request is a stream object, so it will not block but will be piped into the new request effectively as far as I understand :)